module/signature/signer_indices_test.go

```diff
@@ -0,0 +1,681 @@
+		require.NoError(t, err)
+
+		// check verify signer indices
+		unorderedSigners := stakingSigners.Union(beaconSigners) // caution, the Union operation potentially changes the ordering
+		correctEncoding(t, signerIndices, committee, unorderedSigners)
+
+		// check sigTypes
+		canSigners := committeeIdentities.Filter(filter.HasNodeID(unorderedSigners...)).NodeIDs() // generates list of signer IDs in canonical order
+		correctEncoding(t, sigTypes, canSigners, beaconSigners)
+	})
+}
+
+// Test_DecodeSigTypeToStakingAndBeaconSigners uses fuzzy-testing framework Rapid to
+// test the method DecodeSigTypeToStakingAndBeaconSigners:
+// * we generate a set of authorized signer: `committeeIdentities`
+// * part of this set is sampled as staking singers: `stakingSigners`
+// * another part of `committeeIdentities` is sampled as beacon singers: `beaconSigners`
+// * we encode the set and check that the results conform to the protocol specification
+// * We encode the set using `EncodeSignerToIndicesAndSigType` (tested before) and then decode it.
+//   Thereby we should recover the original input. Caution, the order might be different,
+//   so we sort both sets.
+func Test_DecodeSigTypeToStakingAndBeaconSigners(t *testing.T) {
+	rapid.Check(t, func(t *rapid.T) {
+		// select total committee size, number of random beacon signers and number of staking signers
+		committeeSize := rapid.IntRange(1, 272).Draw(t, "committeeSize").(int)
+		numStakingSigners := rapid.IntRange(0, committeeSize).Draw(t, "numStakingSigners").(int)
+		numRandomBeaconSigners := rapid.IntRange(0, committeeSize-numStakingSigners).Draw(t, "numRandomBeaconSigners").(int)
+
+		// create committee
+		committeeIdentities := unittest.IdentityListFixture(committeeSize, unittest.WithRole(flow.RoleConsensus)).Sort(order.Canonical)
+		committee := committeeIdentities.NodeIDs()
+		stakingSigners, beaconSigners := sampleSigners(committee, numStakingSigners, numRandomBeaconSigners)
+
+		// encode
+		signerIndices, sigTypes, err := signature.EncodeSignerToIndicesAndSigType(committee, stakingSigners, beaconSigners)
+		require.NoError(t, err)
+
+		// decode
+		decSignerIdentites, err := signature.DecodeSignerIndicesToIdentities(committeeIdentities, signerIndices)
+		require.NoError(t, err)
+		decStakingSigners, decBeaconSigners, err := signature.DecodeSigTypeToStakingAndBeaconSigners(decSignerIdentites, sigTypes)
+		require.NoError(t, err)
+
+		// verify; note that there is a slightly different convention between Filter and the decoding logic:
+		// Filter returns nil for an empty list, while the decoding logic returns an instance of an empty slice
+		sigIdentities := committeeIdentities.Filter(filter.Or(filter.HasNodeID(stakingSigners...), filter.HasNodeID(beaconSigners...))) // signer identities in canonical order
+		if len(stakingSigners)+len(decBeaconSigners) == 0 {
+			require.Empty(t, decSignerIdentites)
+		} else {
+			require.Equal(t, sigIdentities, decSignerIdentites)
+		}
```

Comment:
```suggestion
		if len(stakingSigners)+len(decBeaconSigners) > 0 {
			require.Equal(t, sigIdentities, decSignerIdentites)
		}
```
Condensing to one branch is a bit clearer for me